xdem/coreg/affine.py

```diff
@@ -473,93 +454,113 @@ def _fit_func(
+            ref = "raster"
+
+        # Pre-process point data
+        point_elev = point_elev.dropna(how="any", subset=[z_name])
+        bounds, resolution = _transform_to_bounds_and_res(rst_elev.shape, transform)
 
-        ref_dem = ref_dem.dropna(how="any", subset=["E", "N", z_name])
-        bounds, resolution = _transform_to_bounds_and_res(tba_dem.shape, transform)
-        points: dict[str, NDArrayf] = {}
         # Generate the x and y coordinates for the TBA DEM
-        x_coords, y_coords = _get_x_and_y_coords(tba_dem.shape, transform)
+        x_coords, y_coords = _get_x_and_y_coords(rst_elev.shape, transform)
         centroid = (np.mean([bounds.left, bounds.right]), np.mean([bounds.bottom, bounds.top]), 0.0)
         # Subtract by the bounding coordinates to avoid float32 rounding errors.
         x_coords -= centroid[0]
         y_coords -= centroid[1]
 
-        gradient_x, gradient_y = np.gradient(tba_dem)
+        gradient_x, gradient_y = np.gradient(rst_elev)
 
         # This CRS is temporary and doesn't affect the result. It's just needed for Raster instantiation.
         dem_kwargs = {"transform": transform, "crs": rio.CRS.from_epsg(32633), "nodata": -9999.0}
         normal_east = Raster.from_array(np.sin(np.arctan(gradient_y / resolution)) * -1, **dem_kwargs)
         normal_north = Raster.from_array(np.sin(np.arctan(gradient_x / resolution)), **dem_kwargs)
         normal_up = Raster.from_array(1 - np.linalg.norm([normal_east.data, normal_north.data], axis=0), **dem_kwargs)
 
-        valid_mask = ~np.isnan(tba_dem) & ~np.isnan(normal_east.data) & ~np.isnan(normal_north.data)
+        valid_mask = ~np.isnan(rst_elev) & ~np.isnan(normal_east.data) & ~np.isnan(normal_north.data)
 
-        points["tba"] = np.dstack(
+        points: dict[str, NDArrayf] = {}
+        points["raster"] = np.dstack(
             [
                 x_coords[valid_mask],
                 y_coords[valid_mask],
-                tba_dem[valid_mask],
+                rst_elev[valid_mask],
                 normal_east.data[valid_mask],
                 normal_north.data[valid_mask],
                 normal_up.data[valid_mask],
             ]
         ).squeeze()
 
-        if any(col not in ref_dem for col in ["nx", "ny", "nz"]):
+        # TODO: Should be a way to not duplicate this column and just feed it directly
+        point_elev["E"] = point_elev.geometry.x.values
+        point_elev["N"] = point_elev.geometry.y.values
+
+        if any(col not in point_elev for col in ["nx", "ny", "nz"]):
             for key, raster in [("nx", normal_east), ("ny", normal_north), ("nz", normal_up)]:
                 raster.tags["AREA_OR_POINT"] = "Area"
-                ref_dem[key] = raster.interp_points(
-                    ref_dem[["E", "N"]].values, shift_area_or_point=True, mode="nearest"
+                point_elev[key] = raster.interp_points(
+                    point_elev[["E", "N"]].values,
+                    shift_area_or_point=True,
```

Comment:
Why did argument "mode" disappear? Btw should we fix it (to bilinear I assume here) or leave it as an optional argument?

Reply:
The "mode" argument does not exist anymore, it's "method" now to be consistent with SciPy interpn. And actually it did not do anything before, it went into the void! (defaulting to bilinear due to the "order" argument that defaulted to one)
So now it's still bilinear everywhere... For easy use, we might want to add a `geoutils.config` argument for users to change it module-wide!

Reply:
Opened https://github.com/GlacioHack/geoutils/issues/530

Reply:
Haha ok that's good this is fixed now ! 🤦 
And good idea to use the config file in the future !